App crashes when listing blank forms from Google Drive without API auth

#### Software and hardware versions 
Current master branch

#### Problem description
App crashes when I try to open my google drive

#### Steps to reproduce the problem
1. In "General Settings" set platform to "Google Drive, Google Sheets"
2. Click on "Get Blank Form" from Main menu

#### Other information 

> Caused by: java.lang.RuntimeException: Can't create handler inside thread that has not called Looper.prepare()
                                                       at android.os.Handler.<init>(Handler.java:200)
                                                       at android.os.Handler.<init>(Handler.java:114)
                                                       at android.app.Dialog.<init>(Dialog.java:113)
                                                       at android.app.AlertDialog.<init>(AlertDialog.java:125)
                                                       at android.app.AlertDialog$Builder.create(AlertDialog.java:968)
                                                       at org.odk.collect.android.activities.GoogleDriveActivity.createAlertDialog(GoogleDriveActivity.java:529)
                                                       at org.odk.collect.android.activities.GoogleDriveActivity.access$500(GoogleDriveActivity.java:96)
                                                       at org.odk.collect.android.activities.GoogleDriveActivity$RetrieveDriveFileContentsAsyncTask.doInBackground(GoogleDriveActivity.java:797)
                                                       at org.odk.collect.android.activities.GoogleDriveActivity$RetrieveDriveFileContentsAsyncTask.doInBackground(GoogleDriveActivity.java:766)

--- a/collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java
+++ b/collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java
@@ -794,7 +794,12 @@ public class GoogleDriveActivity extends ListActivity implements
                 return null;
             } catch (IOException e) {
                 Timber.e(e);
-                createAlertDialog(getString(R.string.google_auth_io_exception_msg));
+                runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        createAlertDialog(getString(R.string.google_auth_io_exception_msg));
+                    }
+                });
             }
             if (rootId == null) {
                 Timber.e("Unable to fetch drive contents");

